Common/Product/SharedProject/HierarchyNode.cs

```diff
@@ -1567,7 +1567,6 @@ internal abstract class HierarchyNode :
             firstChild = null;
             lastChild = null;
             nextSibling = null;
-            itemNode = null;
```

Comment:
This change was added in order to address our memory leaks #871 but is no longer required. For 1.2, it may cause additional problems so I think reverting this line is the correct solution